src/btree/bt_split.c

```diff
@@ -1474,6 +1476,25 @@ __split_multi_inmem(
 		WT_ILLEGAL_VALUE_ERR(session);
 		}
 
+		/*
+		 * If an update from the list was saved into the page image, it
+		 * must have been globally visible and we don't need two copies.
+		 */
+		if (supd->onpage_upd != NULL) {
+			for (prev_upd = upd; prev_upd != NULL &&
+			    prev_upd->next != supd->onpage_upd;
+			    prev_upd = prev_upd->next)
+				;
+			if (prev_upd != NULL) {
+				WT_ASSERT(session,
+				    prev_upd->next == supd->onpage_upd);
+				__wt_update_obsolete_free(
+				    session, page, prev_upd->next);
+				prev_upd->next = NULL;
+			}
+		}
+	}
```

Comment:
We've followed up on this separately: it is only okay to strip updates after the on-page update when reconciliation with `WT_REC_VISIBLE_ALL`, which is now required for update/restore eviction.  Otherwise, since modify operations are not idempotent, we can't re-read the update that was chosen as the on-page version.

I'll look into stripping out updates from `__rec_txn_read` again in other cases: I had a more comprehensive version of that code earlier in this branch but dropped it to avoid complicating review / debugging.